src/db/migrations/20250317184212_add-user-announcements-table.js

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export function up(knex) {
+  return knex.schema
+    .alterTable("announcements", (table) => {
+      table.unique(["announcement_id"]);
+    })
+    .then(() => {
+      return knex.schema.createTable("user_announcements", (table) => {
+        table.increments("id").primary();
+        table.integer("user_id").notNullable();
+        table
+          .string("announcement_id")
+          .references("announcements.announcement_id")
+          .notNullable()
+          .onDelete("CASCADE")
+          .onUpdate("CASCADE");
+        table.string("status").notNullable(); // seen, cleared, clicked, new
+        table.timestamp("seen_at").nullable();
+        table.timestamp("cleared_at").nullable();
+        table.boolean("is_history").defaultTo(false);
```

Comment:
not seeing this in the ticket, what's the purpose of this again?

Reply:
Ah, would it essentially serve the same function as `status`?

I would caution against storing duplicated statuses, since now when you are doing any modification of the column, you'd have to remember to also update this other column.

The best way is to not have this and just have a simple logic in the app to derive `is_history`